operator/controllers/btpoperator_controller.go

```diff
@@ -95,25 +103,236 @@ var (
+		}
+		if err != nil {
+			return err
+		}
+
+		if err := r.deleteOrphanedResources(ctx, configMap); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) createDefaultNamespaceIfNeeded(ctx context.Context) error {
+	namespace := &corev1.Namespace{}
+	namespace.Name = ChartNamespace
+	err := r.Get(ctx, client.ObjectKeyFromObject(namespace), namespace)
+	if k8serrors.IsNotFound(err) {
+		err = r.Create(ctx, namespace)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) buildBtpManagerConfigMap() *corev1.ConfigMap {
+	configMap := &corev1.ConfigMap{}
+	configMap.Namespace = ChartNamespace
+	configMap.Name = btpManagerConfigMap
+	return configMap
+}
+
+func (r *BtpOperatorReconciler) storeChartDetails(ctx context.Context, configMap *corev1.ConfigMap) (bool, error) {
+	newChartVersion, err := ymlutils.ExtractStringValueFromYamlForGivenKey(fmt.Sprintf("%s/Chart.yaml", ChartPath), "version")
+	if err != nil {
+		return false, err
+	}
+
+	newGvks, err := ymlutils.GatherChartGvks(ChartPath)
+	if err != nil {
+		return false, err
+	}
+
+	err = r.createDefaultNamespaceIfNeeded(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if configMap == nil {
+		configMap = r.buildBtpManagerConfigMap()
+		err = r.handleInitialConfigMap(ctx, configMap, newChartVersion, newGvks)
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	} else {
+		return r.handleExistingConfigMap(ctx, configMap, newChartVersion, newGvks)
+	}
+}
+
+func (r *BtpOperatorReconciler) handleInitialConfigMap(ctx context.Context, configMap *corev1.ConfigMap, newChartVersion string, newGvks []schema.GroupVersionKind) error {
+	logger := log.FromContext(ctx)
+	logger.Info("btp manager config dosent exists, new will be created")
+
+	configMap.Data = make(map[string]string)
+
+	newGvksAsStr, err := gvksutils.GvksToStr(newGvks)
+	if err != nil {
+		return err
+	}
+
+	r.setBtpManagerConfigMap(ctx, configMap, newChartVersion, newChartVersion, newGvksAsStr, newGvksAsStr)
+
+	if err := r.Create(ctx, configMap); err != nil {
+		return err
+	}
+
+	r.currentVersion = newChartVersion
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) handleExistingConfigMap(ctx context.Context, configMap *corev1.ConfigMap, newVersion string, newGvks []schema.GroupVersionKind) (bool, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("btp manager config exists")
+
+	currentVersion, ok := configMap.Data[currentCharVersionKey]
+
+	if !ok {
+		return false, fmt.Errorf("%s should be present in configmap but it is not", currentCharVersionKey)
+	}
+
+	currentGvksAsText, ok := configMap.Data[currentGvksKey]
+	if !ok {
+		return false, fmt.Errorf("%s should be present in configmap but it is not", currentGvksAsText)
+	}
+
+	newGvksAsText, err := gvksutils.GvksToStr(newGvks)
+	if err != nil {
+		return false, err
+	}
+
+	if newVersion != currentVersion {
```

Comment:
I suggest to move this condition somewhere at the beginning of this method, if the version didn't change then we don't have to update the configmap. Fetching existing GVKs and creating new data for configmap seems pointless in such case